symmetric-db/src/main/java/org/jumpmind/db/platform/mssql/MsSql2005DdlBuilder.java

```diff
@@ -87,13 +88,24 @@ protected void dropColumnChangeDefaults(Table sourceTable, Column sourceColumn,
 
     @Override
     public String getSqlType(Column column) {
+        Map<String, String> env = System.getenv();
+        boolean useVarcharForText = "y".equalsIgnoreCase(env.get("USE_VARCHAR_FOR_TEXT"));
         String sqlType = super.getSqlType(column);
         if (column.getMappedTypeCode() == Types.VARBINARY && column.getSizeAsInt() > 8000) {
             sqlType = "VARBINARY(MAX)";
         } else if (column.getMappedTypeCode() == Types.VARCHAR && column.getSizeAsInt() > 8000) {
             sqlType = "VARCHAR(MAX)";
         } else if (column.getMappedTypeCode() == Types.DECIMAL && column.getSizeAsInt() > 38) {
             sqlType = String.format("DECIMAL(38,%d)", column.getScale());
+        } else if (useVarcharForText && (column.getMappedTypeCode() == Types.LONGVARCHAR || column.getMappedTypeCode() == Types.LONGNVARCHAR || column
+                .getMappedTypeCode() == Types.CLOB)) {
+            sqlType = "";
+            if (column.getMappedTypeCode() == Types.LONGNVARCHAR) {
+                sqlType = "N";
+                final String newSize = Integer.toString(column.getSizeAsInt() * 2);
+                column.setSize(newSize);
```

Comment:
What is the reason for modifying the column size here?  The getSqlType() method is a get method that normally wouldn't modify what's passed in.  It already returns VARCHAR(MAX) with a maximum size, so there doesn't seem to be a reason to modify the column size also.

Reply:
Sorry, that was incorrect to add. I have removed the lines for newSize.